### PR TITLE
Sb update initial

### PR DESCRIPTION
**Pull request by ScottBailey**
# Changes

Changes:
- Initial import from conda-forge.
- Modifications to build in Anaconda environment.

# Review Information

## Source

https://github.com/mamba-org/mamba/tree/2022.02.28

## License

https://github.com/mamba-org/mamba/blob/2022.02.28/LICENSE

BSD-3

## Upstream Changes

https://github.com/mamba-org/mamba/releases
This is an initial import.

## Issues

https://github.com/mamba-org/mamba/issues
This is an initial import and the upstream is immature - there are going to be issues.

## Pins and Requireds

A number of dependency packages needed to be updated for this. Those packages are pinned to the updated version.

https://github.com/mamba-org/mamba/blob/2022.02.28/pyproject.toml
pybind11 and setuptools.

https://github.com/mamba-org/mamba/blob/2022.02.28/mamba/setup.py

## Testing

Basic conda forge recipes pass.

# Closing Comments

Basic tests pass. Users are theoretically warned this is a bleeding edge tool, so should expect possible issues. Let's take it.

I'd prefer to spend some time working on the upstream project before
incorporating the project this way; however, we have made a decision
to incorporate this package as soon as possible and we should respect
that position.

### CODE REVIEW

**Comment by jezdez** [comment]
@ScottBailey Thanks for all your work, this is really appreciated!

A few notes:
While we're running the experiment with the conda-libmamba-solver package (to gain the confidence with real world usage), we shouldn't ship the mamba CLI on defaults since that would make it much harder for us to support two package managers for our users. 

Since Mamba has its own release workflow on conda-forge, that is an easy work-around for users interested to use the mamba CLI directly.

If and when we add the mamba CLI to defaults itself should basically be not a by-product of the packaging effort, but a distinct decision to start supporting the mamba CLI directly.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -62,34 +76,34 @@ outputs:
 
   - name: libmambapy
     version: {{ libmambapy_version }}
-    script: build_mamba.sh                       # [unix]
-    script: build_mamba.bat                      # [win]
+    script: build_libmambapy.sh                   # [unix]
+    script: bld_libmamba.bat                      # [win]
     build:
       string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:89]: We should skip `py<37` for `libmambapy`, https://github.com/mamba-org/mamba/blob/4b663ada019715ad4c989a1a57e81fe12edde3d8/libmambapy/setup.cfg#L17

File: recipe/meta.yaml [hunk 2]
```diff
@@ -62,34 +76,34 @@ outputs:
...
+        - pybind11 >=2.2
+        #- openssl
+        #- yaml-cpp
+        #- spdlog-fmt-embed
         - cpp-filesystem >=1.5.8
         - termcolor-cpp
         - nlohmann_json
+        - cli11
+        - winreg                                 # [win]
         - {{ pin_subpackage('libmamba', exact=True) }}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:113]: There are missing host dependencies `setuptools` and `wheel`

File: recipe/meta.yaml [hunk 3]
```diff
@@ -104,20 +118,23 @@ outputs:
...
+        - python
       host:
         - python
         - pip
         - openssl
+        - setuptools >=49.2.1
+        - pybind11 >=2.2
+        #- spdlog-fmt-embed
+        #- cli11
         - {{ pin_subpackage('libmambapy', exact=True) }}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: I think there is a missing host dependency `wheel`

File: recipe/meta.yaml [hunk 4]
```diff
@@ -151,6 +167,7 @@ about:
   summary: A fast drop-in alternative to conda, using libsolv for dependency resolution
   description: A fast drop-in alternative to conda, using libsolv for dependency resolution
   dev_url: https://github.com/mamba-org/mamba
+  doc_url:
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: There is a missing doc_url: `https://mamba.readthedocs.io/en/latest/`

File: recipe/meta.yaml (hunk 3 above)
Comment (ScottBailey) [reply recipe/meta.yaml]: This section (`mamba`) is removed.

File: recipe/meta.yaml (hunk 1 above)
Comment (ScottBailey) [reply recipe/meta.yaml:89]: Added.

File: recipe/meta.yaml (hunk 2 above)
Comment (ScottBailey) [reply recipe/meta.yaml:113]: Added.

File: recipe/meta.yaml (hunk 4 above)
Comment (ScottBailey) [reply recipe/meta.yaml]: updated.

**Review by skupr-anaconda — APPROVED** [review]

**Review by antkneed — APPROVED** [review]
👍

**Pull request merged**